instr.c

```diff
@@ -1,6 +1,6 @@
 /*
   Copyright 2013 Google LLC All rights reserved.
-  Copyright 2020 Keegan S. <keegan@sdf.org>
+  Copyright 2020 Keegan Saunders <keegan@undefinedbehaviour.org>
```

Comment:
Looks good to me. Does it help performance any? I've had a few thoughts below on how you might squeeze a little bit more performance out of it.